src/libpmem2/x86_64/memcpy/memcpy_nt_movdir64b.c

```diff
@@ -0,0 +1,385 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2022, Intel Corporation */
+
+#include <immintrin.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "pmem2_arch.h"
+#include "avx.h"
+#include "flush.h"
+#include "memcpy_memset.h"
+#include "memcpy_movdir64b.h"
+#include "valgrind_internal.h"
+
+static force_inline void
+movdir64b(char *dest, const char *src)
+{
+	_movdir64b(dest, src);
+	compiler_barrier();
+}
+
+static force_inline void
+memmove_movnt32x64b(char *dest, const char *src)
+{
+	movdir64b(dest, src);
+	movdir64b(dest + 1 * 64, src + 1 * 64);
```

Comment:
do we need this?
for avx we needed this to use multiple registers. In the case of movdir64b I'm not sure if it gives any performance gain.

Reply:
:+1: